### PR TITLE
LLMeval

### PR DESCRIPTION
**Pull request by vnikouliNLE**
-  same naming of llms in generators and llmeval
- updated documentation
- added sample level match/recall etc., storing them in eval_dev_out.json
- writing all extra sample-level scores in eval_dev_out.json as well

### CODE REVIEW

File: bergen.py [hunk 1]
```diff
@@ -19,8 +19,11 @@ def main(config):
 
     if 'train' in config:
         rag.train()
-
-    rag.eval(dataset_split='dev')
+    if 'dataset_split' in config:
```
Comment (sclincha) [inline bergen.py:22]: What does this have to do with LLM-eval?

Comment (DRRV) [reply bergen.py:22]: you can perform eval on train.  I do it for WTR.  you can alos do it on 'test' !

File: eval.py [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
 import gc
+import pandas as pd
+pd.set_option("display.precision", 4)
 
 class Evaluate:
     @staticmethod
-    def eval(experiment_folder="experiments/", split="dev", bem=False, llm=None, llm_ollama=None, vllm=None,gpt=None,bem_batch_size=1, lid=None, lid_advanced=None, llm_batch_size=1, llm_prompt="default_qa", ollama_url=None, folder=None, force=False, samples=None):
-        def eval_single(experiment_folder, folder, split, model, metric_name, nb_samples=None):
+    def eval(experiment_folder="experiments/", split="dev", bem: bool=False, llm: list[str]=None, llm_ollama: list[str]=None, vllm: list[str]=None, gpt: bool=None, bem_batch_size: int=1, lid: bool=None, lid_advanced: bool=None, llm_batch_size: int=None, llm_prompt: str = "default_qa", ollama_url: str=None, folder: str=None, force: bool=False, samples: int=-1):
+        def eval_single(experiment_folder, folder, split: str, model, metric_name: str, nb_samples: int =-1):
```
Comment (maxime-louis) [inline eval.py:16]: In the env the 'Evaluate' class is not really a class it should be refactored either:
- as a proper class which uses class benefits
- just a set of functions
I added that to the TODOs on slack
